sdk/resourcemanager/Azure.ResourceManager.Core/src/Generated/Models/TagsOperations.cs

```diff
@@ -0,0 +1,432 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// The tag client.
+    /// </summary>
+    public class TagsOperations : OperationsBase
+    {
+        /// <summary> Initializes a new instance of the <see cref="TagsOperations"/> class for mocking. </summary>
+        protected TagsOperations()
+        {
+        }
+
+        internal TagsOperations(OperationsBase options, ResourceIdentifier id) : base(options, id)
+        {
+            _clientDiagnostics = new ClientDiagnostics(ClientOptions);
+        }
+
+        private readonly ClientDiagnostics _clientDiagnostics;
+
+        private string _subscriptionId
+        {
+            get {
+                string subscriptionId;
+                Id.TryGetSubscriptionId(out subscriptionId);
+                return subscriptionId;
+            }
+        }
+
+        /// <summary> Represents the REST operations. </summary>
+        private TagsRestOperations RestClient => new TagsRestOperations(_clientDiagnostics, Pipeline, _subscriptionId, BaseUri);
+
+        /// <summary> Typed Resource Identifier for the container. </summary>
+        public new ResourceIdentifier Id => base.Id as ResourceIdentifier;
+
+        /// <summary> Gets the valid resource type for this object. </summary>
+        protected override ResourceType ValidResourceType => Id.ResourceType;
+
+        /// <summary>
+        /// Create or update tags with the resource.
+        /// </summary>
+        /// <param name="parameters"> The tags to create or update. </param>
+        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
+        /// <returns> The created or updated tags. </returns>
+        public virtual Response<TagsResource> CreateOrUpdateAtScope(TagsResourceData parameters, CancellationToken cancellationToken = default)
```

Comment:
the method name should not contain "AtScope" if the scope is restricted to the Id of the current instance

Reply:
Done